text_test.go

```diff
@@ -137,4 +137,52 @@ func TestExpander(t *testing.T) {
 			So(res.String(), ShouldEqual, "https://github.com/search?q=foo/bar/baz/")
 		})
 	})
+	Convey("Given 'wc/1/2/3/four'", t, func() {
+	    c, _ := loadTestYaml()
+	    l := tokenize("wc/1/2/3/four")
+	    var res bytes.Buffer
+	    res.WriteString(httpsPrefix)
+
+        expandPath(c, l.Front(), &res)
+
+        Convey("result should equal 'https://wildcard.com/1/2/3/4'", func() {
+            So(res.String(), ShouldEqual, "https://wildcard.com/1/2/3/4")
+        })
+	})
```

Comment:
can you add a test for "wc/1/*/3/four" ? Since the wildcard is a [legal url token](https://stackoverflow.com/questions/1547899/which-characters-make-a-url-invalid) I'm curious how this will affect zap.

Reply:
Mmm, good point. Will add a test, but off the top of my head I'd say the changes here would make it impossible to expand path elements of `*` but wouldn't affect URLs with `*` in them.

Reply:
Well that was wrong. Digging further now.

Reply:
Hmm, looks like this isn't really a special case. If a URL contains any of the keys as one of its path elements right now things fail; for example, after adding a test for `ak/expand/j`, this was the result:
```
  Line 209:
  Expected: 'https://kafka.apache.org/expand/j'
  Actual:   'https://kafka.apache.org'
  (Should be equal)
  Diff:     'https://kafka.apache.org/expand/j'
```

with an additional error message of `error in config, no key matching 'expand', 'query' or 'port' in "kafka.apache.org"` printed to the console.

Reply:
A simple check to prevent `*` path elements from being expanded would prevent exceptions in the code path and provide what I think is intuitive functionality for users. One question with that approach is whether checks for `expand`, `query`, etc. should also be included, though. Thoughts on that approach in general?

Reply:
Hmm.... If it's a simple fix then might as well whitelist the special tokens. If it's causes code bloat I'm also happy having the `expand` and `query` as reserved.

Reply:
I've added a fix that causes reserved keywords to be ignored when present in URLs and a few tests to verify it. LMK what you think

Reply:
Perfect, thanks!